test/server/cards/01_SOR/events/AsteroidSanctuary.spec.js

```diff
@@ -0,0 +1,68 @@
+describe('Asteroid Sanctuary', function() {
+    integration(function() {
+        describe('Asteroid Sanctuary\'s ability -', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['asteroid-sanctuary'],
+                        groundArena: ['viper-probe-droid', 'death-star-stormtrooper', 'wampa']
+                    },
+                    player2: {
+                        groundArena: ['death-trooper', 'superlaser-technician']
+                    }
+                });
+            });
+
+            it('should exhaust an enemy unit and give a friendly unit that costs 3 or less a shield', function() {
+                this.player1.clickCard(this.asteroidSanctuary);
+                expect(this.player1).toBeAbleToSelectExactly([this.deathTrooper, this.superlaserTechnician]);
+
+                this.player1.clickCard(this.deathTrooper);
+                expect(this.player1).toBeAbleToSelectExactly([this.viperProbeDroid, this.deathStarStormtrooper]);
+
+                this.player1.clickCard(this.viperProbeDroid);
+                expect(this.deathTrooper.exhausted).toBe(true);
+                expect(this.viperProbeDroid).toHaveExactUpgradeNames(['shield']);
+                expect(this.asteroidSanctuary).toBeInLocation('discard');
```

Comment:
Generally we don't need to test this, we can assume that event resolution is correctly putting the card in discard. If the goal is to check that the card is done resolving, could do `expect(this.player2).toBeActivePlayer()`

Reply:
I opted to include the `toBeActivePlayer` checks to confirm there are no additional actions need from the player. Done in https://github.com/SWU-Karabast/forceteki/pull/88/commits/a726380c1f2d950046f1404d41725e69d45ac4a1